pkg/remoteclient/remoteclient.go

```diff
@@ -289,6 +295,14 @@ func (b *builder) RESTConfig() (*rest.Config, error) {
 		}
 	}
 
+	if override := b.cd.Spec.ControlPlaneConfig.APIServerIPOverride; override != "" {
+		dialer := &net.Dialer{
```

Comment:
nts: net.Dialer implements DialContext -- that's the prod path. We define the interface so we can swap out the impl for UT.